tests/Pickle.py

```diff
@@ -0,0 +1,14 @@
+import pickle
+import unittest
+
+import github
+
+REPO_NAME = "PyGithub/PyGithub"
+
+
+class TestPickle(unittest.TestCase):
+    def test_pickle_github(self):
+        gh = github.Github()
+        gh2 = pickle.loads(pickle.dumps(gh))
+        self.assertIsInstance(gh2, github.Github)
+        self.assertIsNotNone(gh2._Github__requester._Requester__connection_lock)
```

Comment:
Can we test pickling a `GithubObject` as well?
```suggestion
        self.assertIsNotNone(gh2._Github__requester._Requester__connection_lock)

    def test_pickle_repository(self):
        gh = github.Github()
        repo = gh.get_repo(REPO_NAME)

        repo2 = pickle.loads(pickle.dumps(repo))
        self.assertIsInstance(repo2, Repository)
        self.assertIsNotNone(repo2._requester._Requester__connection_lock)
```
You can reuse replay data from `Requester.testBaseUrlPrefixRedirection.txt`:

    tail -n 10 tests/ReplayData/Requester.testBaseUrlPrefixRedirection.txt > tests/ReplayData/Pickle.test_pickle_repository.txt

Reply:
added in https://github.com/PyGithub/PyGithub/pull/2724/commits/63956924eaab9ab233bff9b6503c2275d85596cb